### ISSUE TITLE
using turnTowards with animation

### ISSUE THREAD

**Issue by Sinanaltundag** [Annoraaq/grid-engine#319]
I am new so sorry if its answer simple
I want to turn character and I use "turnTowards" like this 
`this.gridEngine.turnTowards("player", "up-left");`
I can see with `this.gridEngine.getFacingDirection( "player" )`  direction change but character don't turn to same direction on game. 
I could not find any example so can anyone help me

**Comment by Annoraaq** [comment]
Hey @Sinanaltundag, I recently found that to be a bug. I will provide a fix soon.

**Comment by Annoraaq** [comment]
Should be fixed in (just released) version 2.21.1.

**Comment by Sinanaltundag** [comment]
thanks for your support

**Comment by Annoraaq** [comment]
Oh sorry, I did not check that. Do you by any chance know if it is not working for diagonal directions (top-right, down-left, etc) on isometric maps only? Or are diagonal directions not working in general? If you are not sure, don't worry. I will have to check anyway.

**Comment by Sinanaltundag** [comment]
I am not sure but only isometric map problem. because on isometric map any of directions not working

**Comment by Annoraaq** [comment]
Did you base your code on one of the example for isometric movement ([8-direction isometric example](https://annoraaq.github.io/grid-engine/examples/8-directions-isometric.html), [isometric example](https://annoraaq.github.io/grid-engine/examples/isometric.html))?

If so, be aware that these examples use custom animations. That means that the animations are decoupled from GridEngine. That in turn means that you can use `turnTowards(...)` to turn a character, but you will have to update the animation yourself (because GridEngine does not know anything about your custom animation). That does, by the way, not mean that `turnTowards(...)` would have no effect at all in this case. It still changes the `facingDirection` of the character. You can check that using https://annoraaq.github.io/grid-engine/typedoc/classes/GridEngine.html#getFacingDirection.

In case you are using a custom animation like in the two examples, you can take a closer look at the code of those to see how you could adapt your animation. Every time you call `turnTowards` you can call `sprites[charId].setFrame(getStopFrame(direction));`.

That being said, I realized that there is an observable in place called [directionChanged](https://annoraaq.github.io/grid-engine/typedoc/classes/GridEngine.html#directionChanged). It is already used in the example. When I look at it it seems reasonable to also let it emit, when you call directionChanged. I will fix that. In the meantime you can use the workaround I described before.

Lmk if I was not clear enough, so I can elaborate more.

**Comment by Sinanaltundag** [comment]
I use isometric example you are right, I understand problem. I try to create animation, I am newbie so it is hard for me, and if i success I share here.
thanks and if you have any detailed information you can share, I would be grateful

**Comment by Sinanaltundag** [comment]
I success with this line as your help
`this.gridEngine.getSprite("player").setFrame(15);`
thanks again

**Comment by Annoraaq** [comment]
You're welcome. Thanks for pointing out the bug!

I just released another patch 2.21.2. Now `directionChanged()` should also emit when you call `turnTowards()`. So you can adapt your animation like this:

```
this.gridEngine.directionChanged().subscribe(({ charId, direction }) => {
    sprites[charId].setFrame(getStopFrame(direction));
});
```

as shown here:
https://annoraaq.github.io/grid-engine/examples/8-directions-isometric.html
